modules/catalog/src/main/java/org/apache/ignite/internal/catalog/Catalog.java

```diff
@@ -132,11 +132,11 @@ public Collection<CatalogIndexDescriptor> indexes() {
         return indexesById.values();
     }
 
-    public CatalogZoneDescriptor zone(String name) {
+    public @Nullable CatalogZoneDescriptor zone(String name) {
         return zonesByName.get(name);
     }
 
-    public CatalogZoneDescriptor zone(int zoneId) {
+    public @Nullable CatalogZoneDescriptor zone(int zoneId) {
```

Comment:
How it's possible to retrieve zone to a specific point in time? Either specified by timestamp or catalog version or causality token?
How it's possible to retrieve both zones and e.g. tables atomically, meaning for the same catalog version?

Reply:
I think for this we need `org.apache.ignite.internal.catalog.CatalogService`, but it is not yet fully ready and is being improved iteratively.

Reply:
Well the answer occurred to be misleading. Seems that org.apache.ignite.internal.catalog.Catalog is itself a snapshot.
```
    public @Nullable CatalogZoneDescriptor zone(int zoneId, int catalogVersion) {
        return catalog(catalogVersion).zone(zoneId);
    }
```

Reply:
yep